### PR TITLE
[Remove] LegacyESVersion.V_7_2_* and V_7_3_* constants

### PR DESCRIPTION
**Pull request by nknize**
Removes all usages of LegacyESVersion.V_7_2_ and LegacyESVersion.V_7_3 version constants.

relates #2747

### CODE REVIEW

**Comment by nknize** [comment]
The test failure is fixed by #2768 which is ready for review / merge.

**Comment by dblock** [comment]
Needs a rebase, thx.

File: server/src/main/java/org/opensearch/index/engine/ReadOnlyEngine.java [hunk 1]
```diff
@@ -174,25 +172,21 @@ protected void ensureMaxSeqNoEqualsToGlobalCheckpoint(final SeqNoStats seqNoStat
...
-            if (seqNoStats.getMaxSeqNo() != seqNoStats.getGlobalCheckpoint()) {
-                throw new IllegalStateException(
-                    "Maximum sequence number ["
-                        + seqNoStats.getMaxSeqNo()
-                        + "] from last commit does not match global checkpoint ["
-                        + seqNoStats.getGlobalCheckpoint()
-                        + "]"
-                );
-            }
+        assert assertMaxSeqNoEqualsToGlobalCheckpoint(seqNoStats.getMaxSeqNo(), seqNoStats.getGlobalCheckpoint());
```
Comment (reta) [inline server/src/main/java/org/opensearch/index/engine/ReadOnlyEngine.java:181]: Seems like this condition is gone `(seqNoStats.getGlobalCheckpoint() != SequenceNumbers.UNASSIGNED_SEQ_NO)` , on purpose?

Comment (nknize) [reply server/src/main/java/org/opensearch/index/engine/ReadOnlyEngine.java:181]: Correct.. In all modern versions this logic block (checking that the max sequence number is always equal to the global checkpoint) is always executed.  Prior to 7.2 it was only checked if the global checkpoint was was unassigned.

**Review by reta — APPROVED** [review]

**Review by dreamer-89 — APPROVED** [review]

**Comment by nknize** [comment]
Reproducible test failure introduced in #4669 

```
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.opensearch.OpenSearchExceptionTests.testFailureToAndFromXContentWithNoDetails" -Dtests.seed=95EAFFE028A5D654 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=mt -Dtests.timezone=Antarctica/Palmer -Druntime.java=17

org.opensearch.OpenSearchExceptionTests > testFailureToAndFromXContentWithNoDetails FAILED
    org.junit.ComparisonFailure: expected:<...exception, reason=In[valid argument]]> but was:<...exception, reason=In[ternal failure]]>
        at __randomizedtesting.SeedInfo.seed([95EAFFE028A5D654:99BA15775F82EA2C]:0)
        at org.junit.Assert.assertEquals(Assert.java:117)
        at org.junit.Assert.assertEquals(Assert.java:146)
        at org.opensearch.OpenSearchExceptionTests.testFailureToAndFromXContentWithNoDetails(OpenSearchExceptionTests.java:818)
```

Documented and Fixed in this PR.

**Pull request merged**